src/components/Card.js

```diff
@@ -1,46 +1,38 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._data = { name, link };
-    this._name = name;
-    this._link = link;
+  constructor(data, cardSelector, handleImageClick, handleDeleteCard, handleLikeCard) {
+    this._name = data.name;
+    this._link = data.link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteCard = handleDeleteCard;
+    this.id = data._id;
+    this.isLiked = data.isLiked;
+    this._handleLikeCard = handleLikeCard;
   }
 
   _setEventListeners() {
-    //".card__like-button"
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon();
-      });
-    //".card__delete-button"
-    this._cardElement
-      .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
-      });
-    //image preview
-    this._cardImageEl
-      .addEventListener("click", () => {
-        this._handleImageClick({
-          name: this._name,
-          link: this._link,
-        });
+    this._likeButton.addEventListener("click", () => {
+
+      this._handleLikeCard(this);
     });
+
+    this._deleteButton.addEventListener("click", () =>
+      this._handleDeleteCard(this)
+    );
+
+    this._cardImageEl.addEventListener("click", () =>
+      this._handleImageClick(this)
+    );
   }
 
-  _handleDeleteCard() {
-    this._cardElement.remove();
-    this._cardElement = null;
+  handleLikeIcon() {
```

Comment:
Needs correction: you don't need to toggle the likes state, you should set it according to the server response. Please rename this function to `setIsLiked` as the likes state setter. The method will have an argument `isLiked` which you will set and render the button according to it:

```js
// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}
```